index.js

```diff
@@ -31,10 +52,11 @@ module.exports = {
         }
       };
 
-      return new TemplateLinter(tree, {
+      return TemplateLinter.create(tree, {
         annotation: 'TemplateLinter',
         templatercPath: this.project.root + '/.template-lintrc',
-        generateTestFile: this.project.generateTestFile,
+        testGenerator: this._options.testGenerator,
+        group: (this._options.group !== false) ? type : undefined,
```

Comment:
This makes it only possible for end users of ember-cli-template-lint to disable grouping, not to add a custom group. Is that intended?  Based on one of the tests, it seemed that you did mean to allow this to be specified as a string...